src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/dcoff.cpp

```diff
@@ -122,15 +217,48 @@ void apply_remap(Subgraph& fcall, const ClosureRemap& m) {
     // reserve a new_scales vector to have the same size, filled with
     // empty tensors by default.
     for (auto&& i : m.closure_remap) {
-        new_closure.push_back(fcall._closure[i]);
+        // Check if the index is marked for transposition
+        if (std::find(m.transpose_indices.begin(), m.transpose_indices.end(), i) != m.transpose_indices.end()) {
+            // Transpose the tensor before adding it to new_closure
+            new_closure.push_back(pattern_utils::transpose_tensor(fcall._closure[i]));
+        } else {
+            // Add the original tensor to new_closure
+            new_closure.push_back(fcall._closure[i]);
+        }
 
+        // Handle scale remap
         auto scale_iter = m.scale_remap.find(i);
-        new_scales.push_back(scale_iter != m.scale_remap.end() ? fcall._closure[scale_iter->second] : ov::Tensor());
-        // Check for asymmetric zero points and add them to new_zerops
+        if (scale_iter != m.scale_remap.end()) {
+            // Check if the scale index is marked for transposition
+            if (std::find(m.transpose_indices.begin(), m.transpose_indices.end(), scale_iter->second) != m.transpose_indices.end()) {
+                // Transpose the tensor before adding it to new_scales
+                new_scales.push_back(pattern_utils::transpose_tensor(fcall._closure[scale_iter->second]));
+            } else {
+                // Add the original tensor to new_scales
+                new_scales.push_back(fcall._closure[scale_iter->second]);
+            }
+        } else {
+            new_scales.push_back(ov::Tensor());
+        }
+
+        // Handle zero point remap
         auto zerop_iter = m.zerop_remap.find(i);
-        const auto& zerop = zerop_iter != m.zerop_remap.end() ? fcall._closure[zerop_iter->second] : m.zero_points[i];
-        new_zerops.push_back(zerop);
+        if (zerop_iter != m.zerop_remap.end()) {
+            // Check if the zero point index is marked for transposition
+            if (std::find(m.transpose_indices.begin(), m.transpose_indices.end(), zerop_iter->second) != m.transpose_indices.end()) {
+                // Transpose the tensor before adding it to new_zerops
+                new_zerops.push_back(pattern_utils::transpose_tensor(fcall._closure[zerop_iter->second]));
+            } else {
+                // Add the original tensor to new_zerops
+                new_zerops.push_back(fcall._closure[zerop_iter->second]);
+            }
+        } else {
+            // Add the zero point tensor from the closure remap
+            const auto& zerop = m.zero_points[i];
+            new_zerops.push_back(zerop);
+        }
     }
+
```

Comment:
Looks too much, you can form the tensor first and then transpose only the affected once. It may be one more separate loop. Will make the code clear.